docker-compose-with-images.yaml

```
@@ -0,0 +1,54 @@
+# SPDX-FileCopyrightText: 2023 The Aalto Grades Developers
+#
+# SPDX-License-Identifier: MIT
+
+version: "3.7"
+
+services:
+
+  reverse-proxy:
+    build:
+      context: ./reverse-proxy
+    restart: always
+    container_name: reverse-proxy
+    ports:
+    - 8080:80
+    networks:
+    - local
+
+  frontend:
+    image: ghcr.io/aalto-grades/aalto-grades-frontend:deployment-configuration
+    restart: always
+    container_name: frontend-server
+    tty: true
+    networks:
+      - local
+
+  backend:
+    image: ghcr.io/aalto-grades/aalto-grades-backend:deployment-configuration
+    tty: true
+    restart: always
+    container_name: backend-server
+    environment:
+      POSTGRES_URL: "database"
+      POSTGRES_USER: "postgres"
+      POSTGRES_DATABASE: "postgres"
+      AALTO_GRADES_JWT_SECRET: "${AALTO_GRADES_JWT_SECRET}"
+      POSTGRES_PASSWORD: "${POSTGRES_PASSWORD}"
+      SISU_API_KEY: "${SISU_API_KEY}"
+      SISU_API_URL: "${SISU_API_URL}"
+      AALTO_GRADES_FRONTEND_CORS_ORIGIN: "https://https://aalto-grades.cs.aalto.fi"
+    networks:
+      - local
+
+  database:
+    image: postgres
+    restart: always
+    environment:
+      POSTGRES_PASSWORD: "${POSTGRES_PASSWORD}"
+    networks:
+      - local

```

This is identical to the common database service, so it should be extended instead:
```suggestion
  database:
    extends:
      file: ./common-services.yaml
      service: database
```